File: backend/models/propertyModel.js
const mongoose = require("mongoose");

const propertySchema = new mongoose.Schema(
  {
    title: { type: String, required: true },
    description: { type: String },
    type: { type: String, enum: ["sale", "rent"], required: true },
    price: { type: Number, required: true },
    area: { type: Number },
    rooms: { type: Number },
    status: { type: String, enum: ["pending", "approved", "sold", "rented"], default: "pending" },
    city: { type: String },
    images: [{ type: String }],
    agent: { type: mongoose.Schema.Types.ObjectId, ref: "User", required: true }, // reference to Agent
    views: { type: Number, default: 0 },
  },
  { timestamps: true }
);

module.exports = mongoose.model("Property", propertySchema);

File: flutterfront/lib/models/PropertyModel.dart
class Property {
  final String id; 
  final String title;
  final String description;
  final String type; // "sale" أو "rent"
  final double price;
  final double? area;
  final int? rooms;
  final String status; // "pending", "approved", "sold", "rented"
  final String? city;
  final List<String> images;
  final String agentId;
  final int views;

  Property({
    required this.id,
    required this.title,
    required this.description,
    required this.type,
    required this.price,
    this.area,
    this.rooms,
    required this.status,
    this.city,
    required this.images,
    required this.agentId,
    required this.views,
  });
}
